### PR TITLE
Mr/article sidebar

### PR DESCRIPTION
**Pull request by melissarudofsky**
Screen Shot 2020-03-04 at 2 20 58 PM

### CODE REVIEW

File: src/templates/article.js [hunk 1]
```diff
@@ -9,12 +9,48 @@ import ArticleShareFooter from '../components/dev-hub/article-share-footer';
...
 import { getTagLinksFromMeta } from '../utils/get-tag-links-from-meta';
 import { getTagPageUriComponent } from '../utils/get-tag-page-uri-component';
-import { normalizePath } from '../utils/normalize-path';
 import { useSiteMetadata } from '../hooks/use-site-metadata';
+import ShareMenu from '../components/dev-hub/share-menu';
+import ContentsMenu from '../components/dev-hub/contents-menu';
+import Contents from '../../src/components/Contents.js';
+import { getNestedValue } from '../utils/get-nested-value';
+
+const findSectionHeadings = (nodes, key, value) => {
```
Comment (melissarudofsky) [inline src/templates/article.js]: Does anybody know if its possible to have a heading within a heading with our rsts?

File: src/components/dev-hub/contents-menu.js [hunk 2]
```diff
@@ -54,27 +67,32 @@ const Contents = styled('ul')`
  * @property {object[]} props.contents
  */
 
-const ContentsMenu = ({ title, contents = [] }) => {
+const ContentsMenu = ({ title, headingNodes, ...props }) => {
```
Comment (melissarudofsky) [inline src/components/dev-hub/contents-menu.js]: Did we get these ContentsMenu styles from anywhere?

Comment (jbouzi12) [reply src/components/dev-hub/contents-menu.js]: They are based on the designs if thats what you mean?

Comment (melissarudofsky) [reply src/components/dev-hub/contents-menu.js]: Oo I see it now! Thanks!

Comment (melissarudofsky) [reply src/components/dev-hub/contents-menu.js]: going to update the styles

**Comment by jestapinski** [comment]
They might want subheadings, I know they were told to use h2/h3 for headings but not sure how they would want subheadings to look. I can imagine just top level headings would be fine for now?

Not sure if anyone has tried to do it, but would not be surprised.

**Comment by melissarudofsky** [comment]
> I'm not sure if there could be a heading within a heading in the Rst. Are there any articles we can check out locally that would have this menu implemented?

all of the articles should work!

**Comment by jbouzi12** [comment]
> > I'm not sure if there could be a heading within a heading in the Rst. Are there any articles we can check out locally that would have this menu implemented?
> 
> all of the articles should work!

Cool I'll take a look now. The share menu looks great though!

File: src/templates/article.js [hunk 3]
```diff
@@ -141,22 +173,35 @@ const Article = props => {
...
-
+            <Container>
+                <Icons>
+                    <ContentsMenu
+                        title="Contents"
+                        headingNodes={headingNodes}
+                        height={size.default}
+                        width={size.default}
+                    />
+                    <ShareMenu
```
[the remark this answers is not in the document]
Comment (melissarudofsky) [reply src/templates/article.js]: hmm that's so weird! just tested it out and it looks fine on my computer. I'll try merging in staging!

[the remark this answers is not in the document]
Comment (melissarudofsky) [reply src/templates/article.js]: I have no idea why this is happening, everything looks fine on my end. Which article were you viewing?

[the remark this answers is not in the document]
Comment (greggb) [reply src/templates/article.js]: Seems to work once I refreshed (http://localhost:8000/article/active-active-application-architectures).

@melissarudofsky When you hover over the link icon do you see the link or the default cursor? Maybe I'm having some weird local issues, but all the other icons show the link cursor and this one shows the default arrow cursor

**Pull request merged**